sketch/hash_base.hpp

```diff
@@ -33,57 +33,47 @@ class HashBase {
 
     void set_hashes_for_testing(const std::vector<std::unordered_map<T, T>> &h) { hashes = h; }
 
-    HashAlgorithm to_hash_algorithm(std::string name) {
-        if (name == "uniform") {
-            hash_algorithm = HashAlgorithm::uniform;
-        } else if (name == "crc32") {
-            hash_algorithm = HashAlgorithm::crc32;
-        }
-        return hash_algorithm;
-    }
-
 
   protected:
     T set_size;
-    size_t sketch_dim;
-    size_t hash_size;
+    size_t sketch_dim {};
+    size_t hash_size {};
```

Comment:
I find `= 0` more readable, as you don't have to know the conventions of uniform initialization. I know some people rave about this, but at the end of the day no-one cares about consistency here. `sketch_dim = 0` will always beat in readability `sketch_dim {}`

Reply:
agreed. But removed this change again since the default constructor is also gone (so now you get a warning when forgetting to initialise them).